src/ShaderUtils.h

```diff
@@ -4,6 +4,8 @@
 #include "OpenGL.h"
 #include "Combiner.h"
 
+static std::string finalString;
```

Comment:
1. finalString is a bad name. Very bad for global name.
2. You don't need to declare it there. Move declaration to addGLSLVersion